crates/kitsune_p2p/kitsune_p2p/src/metrics.rs

```diff
@@ -328,3 +499,68 @@ impl std::fmt::Display for Metrics {
         Ok(())
     }
 }
+
+/// Synchronization primitive around the Metrics struct.
+#[derive(Clone)]
+pub struct MetricsSync(Arc<parking_lot::RwLock<Metrics>>);
```

Comment:
Wondering why you needed an extra mutex around Metrics when they're already accessed from within another mutex?